File: woocommerce_sync/page/woocommerce_config/woocommerce_config.js
frappe.pages['woocommerce-config'].on_page_load = function(wrapper) {
    let page = frappe.ui.make_app_page({
        parent: wrapper,
        title: 'WooCommerce Configuration',
        single_column: true
    });

    $(frappe.render_template("woocommerce_config", {})).appendTo(page.body);

    // Load existing config
    frappe.call({
        method: "woocommerce_sync.woocommerce_sync.page.woocommerce_config.woocommerce_config.get_config",
        callback: function(r) {
            if (r.message) {
                let cfg = r.message;
                $("#woocommerce_url").val(cfg.woocommerce_url || "");
                $("#consumer_key").val(cfg.consumer_key || "");
                $("#consumer_secret").val(cfg.consumer_secret || "");
                $("#enable_sync").prop("checked", cfg.enable_sync ? 1 : 0);
                $("#sync_interval").val(cfg.sync_interval || "Daily");
                $("#last_sync").text(cfg.last_sync || "-");
                $("#sync_status").text(cfg.sync_status || "-");
            }
        }
    });

    // Save config
    $("#save-config").on("click", function() {
        let config = {
            woocommerce_url: $("#woocommerce_url").val(),
            consumer_key: $("#consumer_key").val(),
            consumer_secret: $("#consumer_secret").val(),
            enable_sync: $("#enable_sync").is(":checked") ? 1 : 0,
            sync_interval: $("#sync_interval").val()
        };

        frappe.call({
            method: "woocommerce_sync.woocommerce_sync.page.woocommerce_config.woocommerce_config.save_config",
            args: { config },
            callback: function(r) {
                frappe.show_alert({ message: r.message, indicator: "green" });
            }
        });
    });

    // Test connection
    // $("#test-connection").on("click", function() {
    //     frappe.call({
    //         method: "woocommerce_sync.woocommerce_sync.page.woocommerce_config.woocommerce_config.test_connection",
    //         callback: function(r) {
    //             frappe.msgprint(r.message);
    //         }
    //     });
    // });

    // // Sync now
    // $("#sync-now").on("click", function() {
    //     frappe.call({
    //         method: "woocommerce_sync.woocommerce_sync.page.woocommerce_config.woocommerce_config.sync_now",
    //         callback: function(r) {
    //             frappe.msgprint(r.message);
    //         }
    //     });
    // });
};

File: woocommerce_sync/page/woocommerce_config/woocommerce_config.py
import frappe
from woocommerce import API

# This must be a Single DocType you create (e.g., "WooCommerce Sync Settings")
CONFIG_KEY = "WooCommerce Settings"


def _get_config_doc():
    """Use Frappe's single-value store (tabSingles)."""
    return frappe.db.get_singles_dict(CONFIG_KEY)


@frappe.whitelist()
def get_config():
    """Fetch current WooCommerce configuration"""
    return _get_config_doc()


@frappe.whitelist()
def save_config(config):
    """Save WooCommerce config into Singles"""
    if isinstance(config, str):
        config = frappe.parse_json(config)

    for key, value in config.items():
        frappe.db.set_single_value(CONFIG_KEY, key, value)

    return "✅ Configuration saved successfully!"


@frappe.whitelist()
def test_connection():
    """Test connection to WooCommerce"""
    cfg = _get_config_doc()
    try:
        wcapi = API(
            url=cfg.get("woocommerce_url"),
            consumer_key=cfg.get("consumer_key"),
            consumer_secret=cfg.get("consumer_secret"),
            version="wc/v3"
        )
        response = wcapi.get("products")
        if response.status_code == 200:
            return "✅ Connection successful!"
        return f"❌ Failed: {response.text}"
    except Exception as e:
        return f"❌ Error: {str(e)}"


@frappe.whitelist()
def sync_now():
    """Start a manual sync"""
    frappe.db.set_single_value(CONFIG_KEY, "sync_status", "Running")
    return "🔄 Sync started..."
